pkg/kubectl/cmd/cmd.go

```diff
@@ -125,25 +125,26 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return describer, nil
 		},
-		Printer: func(cmd *cobra.Command, mapping *meta.RESTMapping, noHeaders bool) (kubectl.ResourcePrinter, error) {
+		Printer: func(mapping *meta.RESTMapping, noHeaders bool) (kubectl.ResourcePrinter, error) {
 			return kubectl.NewHumanReadablePrinter(noHeaders), nil
 		},
-		Resizer: func(cmd *cobra.Command, mapping *meta.RESTMapping) (kubectl.Resizer, error) {
+		Resizer: func(mapping *meta.RESTMapping) (kubectl.Resizer, error) {
 			client, err := clients.ClientForVersion(mapping.APIVersion)
 			if err != nil {
 				return nil, err
 			}
 			return kubectl.ResizerFor(mapping.Kind, client)
 		},
-		Reaper: func(cmd *cobra.Command, mapping *meta.RESTMapping) (kubectl.Reaper, error) {
+		Reaper: func(mapping *meta.RESTMapping) (kubectl.Reaper, error) {
 			client, err := clients.ClientForVersion(mapping.APIVersion)
 			if err != nil {
 				return nil, err
 			}
 			return kubectl.ReaperFor(mapping.Kind, client)
 		},
-		Validator: func(cmd *cobra.Command) (validation.Schema, error) {
-			if cmdutil.GetFlagBool(cmd, "validate") {
+		Validator: func() (validation.Schema, error) {
+			validationFlag := flags.Lookup("validate")
+			if validationFlag != nil && validationFlag.Value.String() == "true" {
```

Comment:
Fine with me if you want to extend the scope of the PR to better handle global flags. If you just want this to be cmd refactoring, we should preserve the existing behavior with respect to `validate`, and that pretty much requires adding it to `f.flags`.